src/uidl-definitions/schemas/new-component.json

```diff
@@ -0,0 +1,279 @@
+          "default": "MyComponent"
+        },
+        "node": {
+            "$ref": "#/definitions/node"
+          },
+        "meta": {
+          "type": "object"
+        },
+        "stateDefinitions": {
+          "$ref": "#/definitions/stateDefinitions"
+        },
+        "propDefinitions": {
+          "$ref": "#/definitions/propDefinitions"
+        }
+      },
+      "definitions": {
+        "dependency": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["type"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["package", "local", "library"],
+              "examples": ["package", "local", "library"]
+            },
+            "path": {"type": "string"},
+            "version": {"type": "string", "default": "1.0.0"},
+            "meta": {
+              "type": "object", 
+              "additionalProperties": false,
+              "properties": {
+                "namedImport": {"type": "boolean", "default": false},
+                "originalName": {"type": "string"}
+              }
+            }
+          }
+        },
+        "propDefinitions": {
+          "type": "object",
+          "patternProperties": {
+            ".*": {
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["type"],
+              "properties": {
+                "type": {"type": "string", "enum": ["string", "boolean", "number", "array", "func", "object", "children"]},
+                "defaultValue": {},
+                "meta": {"type": "object"}
+              }
+            }
+          }
+        },
+        "stateDefinitions": {
+          "type": "object",
+          "patternProperties": {
+            ".*": {
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["type", "defaultValue"],
+              "properties": {
+                "type": {"type":"string", "enum": ["string", "boolean", "number", "object", "func", "array", "router"]},
+                "values": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "value": {},
+                      "meta": { 
+                        "type": "object",
+                        "properties": {
+                          "componentName": { "type": "string" },
+                          "path": { "type": "string" },
+                          "fileName": { "type": "string" }
+                        }
+                      },
+                      "transitions": { "type": "object" }
+                    }
+                  }
+                },
+                "defaultValue": {},
+                "actions": {"type": "array"}
+              }
+            }
+          }
+        },
+        "style": {},
+        "node": {
+          "oneOf": [{
```

Comment:
this oneOf is good, but we could also add a separate `string` as a type, if we want to support the old style text nodes, ex: "$props.title". let's talk tomorrow